File: src/main/java/com/project/pantau/mapper/CategoryMapper.java
package com.project.pantau.mapper;

import com.project.pantau.dto.category.CategoryResponse;
import com.project.pantau.entity.Category;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(config = MapStructConfig.class)
public interface CategoryMapper {
    @Mapping(
            target = "isActive",
            expression = "java(category.isActive())"
    )
    CategoryResponse toResponse(Category category);

    List<CategoryResponse> toResponse(List<Category> categories);
}

File: backend/internal/dto/mapper/category_mapper.go
package mapper

import (
	"pantau/internal/dto/category"
	"pantau/internal/entity"
)

func CategoryToResponse(cat *entity.Category) *category.CategoryResponse {
	if cat == nil {
		return nil
	}
	return &category.CategoryResponse{
		ID:       cat.ID,
		Name:     cat.Name,
		Slug:     cat.Slug,
		IsActive: cat.IsActive,
	}
}

func CategoriesToResponse(categories []entity.Category) []category.CategoryResponse {
	if categories == nil {
		return nil
	}
	responses := make([]category.CategoryResponse, len(categories))
	for i := range categories {
		responses[i] = *CategoryToResponse(&categories[i])
	}
	return responses
}
